### PR TITLE
🐛 Fixed author role permission to change author

### PR DESCRIPTION
**Pull request by ErisDS**
Hopefully this has decent enough test coverage that it doesn't have weird edge cases.

This updates the API to reflect the UI.
Author roles can only create posts with their own author_id, and cannot edit the author_id.

no issue

- This bug affects the private / undocumented API only
- Author role users should not be allowed to change the author of a post

### CODE REVIEW

File: core/server/api/posts.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ var Promise = require('bluebird'),
         'created_by', 'updated_by', 'published_by', 'author', 'tags', 'fields',
         'next', 'previous', 'next.author', 'next.tags', 'previous.author', 'previous.tags'
     ],
+    unsafeAttrs = ['author', 'author_id'],
```
Comment (kirrg001) [inline core/server/api/posts.js]: I don't think we need the property `author`, because of https://github.com/TryGhost/Ghost/blob/master/core/server/api/utils.js#L300?

File: core/test/functional/routes/api/posts_spec.js [hunk 2]
```diff
@@ -1321,25 +1321,78 @@ describe('Post API', function () {
...
+                    .expect('Content-Type', /json/)
+                    .expect('Cache-Control', testUtils.cacheRules.private)
+                    .expect(403)
+                    .end(function (err, res) {
+                        if (err) {
+                            return done(err);
+                        }
+
+                        should.exist(res.body.errors);
+                        res.body.errors[0].errorType.should.eql('NoPermissionError');
```
Comment (kirrg001) [inline core/test/functional/routes/api/posts_spec.js]: Should we return a more specific error message?
Because the message says if i modify the author: "You do not have permission to edit posts".

Comment (kirrg001) [reply core/test/functional/routes/api/posts_spec.js]: Hmm or maybe not. Because i am not able to change my author id in the admin panel, there is no UI if i am an author.

And if somebody else changes my id, the message is correct.

**Comment by kirrg001** [comment]
The functionality works 👍  I left two smaller comments.

File: core/test/functional/routes/api/posts_spec.js (hunk 2 above)
Comment (ErisDS) [reply core/test/functional/routes/api/posts_spec.js]: Ideally, yes, it should be more specific. In practice, I can't see a way to do this that doesn't require more refactoring & I don't think it's important enough given that there is no UI.

Something for future TODOs - have added a note to #9043

Comment (kirrg001) [reply core/test/functional/routes/api/posts_spec.js]: Agree 👍

**Comment by ErisDS** [comment]
Removed the author attr - you're correct. All green.

**Review by kirrg001 — APPROVED** [review]
